src/components/transactions/BatchExecuteButton/index.tsx

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useContext, useMemo } from 'react'
+import { Button } from '@mui/material'
+import css from './styles.module.css'
+import { BatchExecuteHoverContext } from '@/components/transactions/BatchExecuteButton/BatchExecuteHoverProvider'
+import { Transaction, TransactionListItem } from '@gnosis.pm/safe-react-gateway-sdk'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { isMultisigExecutionInfo, isTransactionListItem } from '@/utils/transaction-guards'
+import { useAppSelector } from '@/store'
+import { selectPendingTxs } from '@/store/pendingTxsSlice'
+import CustomTooltip from '@/components/common/CustomTooltip'
+
+const BATCH_LIMIT = 10
+
+export const getBatchableTransactions = (items: (TransactionListItem | Transaction[])[], nonce: number) => {
+  const batchableTransactions: Transaction[] = []
+  let currentNonce = nonce
+
+  const grouped = items
+    .map((item) => {
+      if (Array.isArray(item)) return item
+      if (isTransactionListItem(item)) return [item]
+    })
+    .filter(Boolean) as Transaction[][]
```

Comment:
Some of the items are nested arrays i.e. transactions with the same nonce so it would filter those out as well. The casting is there to get rid of `undefined` because typescript doesn't infer that.

Reply:
Can we use a combination of guards for the nested arrays? It may infer correctly if you `filter` before the `map` too. Otherwise this is fine to resolve.

Reply:
It is already infered correctly. Maybe we need a better name for the guard since there is also a type called `TransactionListItem` but the guard `isTransactionListItem` actually resolves to `Transaction`.

Reply:
Alright, I just saw the casting and thought we could use guards instead.